From 4f8bc734ab5cb0316d770fc373e1c652b5628e02 Mon Sep 17 00:00:00 2001
From: Olaf Hering <olaf@aepfle.de>
Date: Thu, 28 Mar 2019 18:11:49 +0100
Subject: [PATCH] [libUPnP] Remove unused headers from Platinum/Source/Extras

No code from the Extras directory is compiled, so it is unlikely that
any of the interfaces will be used. Remove the headers and remove
Platinum/Source/Extras from the list of include directories.

Signed-off-by: Olaf Hering <olaf@aepfle.de>
---
 lib/libUPnP/CMakeLists.txt                    |  1 -
 .../Platinum/Source/Platinum/Platinum.h       |  7 -----
 ...nused-headers-from-Platinum-Source-E.patch | 29 +++++++++++++++++++
 xbmc/network/CMakeLists.txt                   |  1 -
 4 files changed, 29 insertions(+), 9 deletions(-)
 create mode 100644 lib/libUPnP/patches/0048-libUPnP-Remove-unused-headers-from-Platinum-Source-E.patch

diff --git a/lib/libUPnP/CMakeLists.txt b/lib/libUPnP/CMakeLists.txt
index 5aa41aa6499d..79499ac3b33e 100644
--- a/lib/libUPnP/CMakeLists.txt
+++ b/lib/libUPnP/CMakeLists.txt
@@ -114,7 +114,6 @@ target_include_directories(upnp PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}
                                        Platinum/Source/Devices/MediaConnect
                                        Platinum/Source/Devices/MediaRenderer
                                        Platinum/Source/Devices/MediaServer
-                                       Platinum/Source/Extras
                                        Neptune/Source/Core
                                        Neptune/Source/System/Posix)
 if(CORE_SYSTEM_NAME STREQUAL windows OR CORE_SYSTEM_NAME STREQUAL windowsstore)
diff --git a/lib/libUPnP/Platinum/Source/Platinum/Platinum.h b/lib/libUPnP/Platinum/Source/Platinum/Platinum.h
index a7f47a624147..970cb8bd957a 100644
--- a/lib/libUPnP/Platinum/Source/Platinum/Platinum.h
+++ b/lib/libUPnP/Platinum/Source/Platinum/Platinum.h
@@ -108,11 +108,4 @@ cross-platform logging system.
 #include "PltXbox360.h"
 #include "PltMediaConnect.h"
 
-#include "PltDownloader.h"
-#include "PltStreamPump.h"
-#include "PltFrameBuffer.h"
-#include "PltFrameServer.h"
-#include "PltFrameStream.h"
-#include "PltRingBufferStream.h"
-
 #endif /* _PLATINUM_H_ */
diff --git a/lib/libUPnP/patches/0048-libUPnP-Remove-unused-headers-from-Platinum-Source-E.patch b/lib/libUPnP/patches/0048-libUPnP-Remove-unused-headers-from-Platinum-Source-E.patch
new file mode 100644
index 000000000000..c1cdb8f95044
--- /dev/null
+++ b/lib/libUPnP/patches/0048-libUPnP-Remove-unused-headers-from-Platinum-Source-E.patch
@@ -0,0 +1,29 @@
+From: Olaf Hering <olaf@aepfle.de>
+Date: Thu, 28 Mar 2019 18:07:31 +0100
+Subject: [libUPnP] Remove unused headers from Platinum/Source/Extras
+
+No code from the Extras directory is compiled, so it is unlikely that
+any of the interfaces will be used. Remove the headers and remove
+Platinum/Source/Extras from the list of include directories.
+
+Signed-off-by: Olaf Hering <olaf@aepfle.de>
+---
+ lib/libUPnP/Platinum/Source/Platinum/Platinum.h | 7 -------
+ 1 file changed, 7 deletions(-)
+
+diff --git a/lib/libUPnP/Platinum/Source/Platinum/Platinum.h b/lib/libUPnP/Platinum/Source/Platinum/Platinum.h
+index a7f47a6241..970cb8bd95 100644
+--- a/lib/libUPnP/Platinum/Source/Platinum/Platinum.h
++++ b/lib/libUPnP/Platinum/Source/Platinum/Platinum.h
+@@ -108,11 +108,4 @@ cross-platform logging system.
+ #include "PltXbox360.h"
+ #include "PltMediaConnect.h"
+ 
+-#include "PltDownloader.h"
+-#include "PltStreamPump.h"
+-#include "PltFrameBuffer.h"
+-#include "PltFrameServer.h"
+-#include "PltFrameStream.h"
+-#include "PltRingBufferStream.h"
+-
+ #endif /* _PLATINUM_H_ */
diff --git a/xbmc/network/CMakeLists.txt b/xbmc/network/CMakeLists.txt
index c8acb897d94c..9e042c6dd5f0 100644
--- a/xbmc/network/CMakeLists.txt
+++ b/xbmc/network/CMakeLists.txt
@@ -62,7 +62,6 @@ if(ENABLE_UPNP)
                                                      ${CMAKE_SOURCE_DIR}/lib/libUPnP/Platinum/Source/Devices/MediaConnect
                                                      ${CMAKE_SOURCE_DIR}/lib/libUPnP/Platinum/Source/Devices/MediaRenderer
                                                      ${CMAKE_SOURCE_DIR}/lib/libUPnP/Platinum/Source/Devices/MediaServer
-                                                     ${CMAKE_SOURCE_DIR}/lib/libUPnP/Platinum/Source/Extras
                                                      ${CMAKE_SOURCE_DIR}/lib/libUPnP/Neptune/Source/System/Posix
                                                      ${CMAKE_SOURCE_DIR}/lib/libUPnP/Neptune/Source/Core)
 endif()
